src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -218,42 +215,39 @@ public synchronized RcaResponse readRca(String rca) {
 
   private void readSummary(GenericSummary upperLevelSummary, int upperLevelPrimaryKey) {
     String upperLevelTable = upperLevelSummary.getTableName();
+    List<Class<? extends GenericSummary>> clazzList = SQLiteQueryUtils.getNestedTableMap().getOrDefault(upperLevelSummary.getClass(), null);
+
     // stop the recursion here if the table does not have any nested summary.
-    if (!SQLiteQueryUtils.getNestedTableMap().containsKey(upperLevelTable)) {
+    if (clazzList == null) {
       return;
     }
-    String currLevelTable = SQLiteQueryUtils.getNestedTableMap().get(upperLevelTable);
-    Field<Integer> foreignKeyField = DSL.field(
-        SQLiteQueryUtils.getPrimaryKeyColumnName(upperLevelTable), Integer.class);
-    SelectJoinStep<Record> rcaQuery = SQLiteQueryUtils
-        .buildSummaryQuery(create, currLevelTable, upperLevelPrimaryKey, foreignKeyField);
-    try {
-      Result<Record> recordList = rcaQuery.fetch();
-      for (Record record : recordList) {
-        GenericSummary summary = null;
-        if (upperLevelSummary instanceof RcaResponse) {
-          summary = HotClusterSummary.buildSummary(record);
-        }
-        else if (upperLevelSummary instanceof HotClusterSummary) {
-          summary = HotNodeSummary.buildSummary(record);
-        }
-        else if (upperLevelSummary instanceof HotNodeSummary) {
-          summary = HotResourceSummary.buildSummary(record);
-        }
-        else if (upperLevelSummary instanceof HotResourceSummary) {
-          summary = TopConsumerSummary.buildSummary(record);
-        }
-        if (summary != null) {
-          Field<Integer> primaryKeyField = DSL.field(
-              SQLiteQueryUtils.getPrimaryKeyColumnName(summary.getTableName()), Integer.class);
-          readSummary(summary, record.get(primaryKeyField));
-          upperLevelSummary.addNestedSummaryList(summary);
+    for (Class<? extends GenericSummary> clazz : clazzList) {
+      Field<Integer> foreignKeyField = DSL.field(
+          SQLiteQueryUtils.getPrimaryKeyColumnName(upperLevelTable), Integer.class);
+      SelectJoinStep<Record> rcaQuery = SQLiteQueryUtils
+          .buildSummaryQuery(create, clazz.getSimpleName(), upperLevelPrimaryKey, foreignKeyField);
+      try {
+        Result<Record> recordList = rcaQuery.fetch();
+        for (Record record : recordList) {
+          Method method = clazz.getMethod("buildSummary", Record.class);
+          GenericSummary summary = (GenericSummary) method.invoke(null, record);
```

Comment:
`(GenericSummary) method.invoke(null, record);`, is the null required here?

Reply:
yes. because buildSummary is a static method